myhpom/tests/test_cloudfactory_tasks.py

```diff
@@ -94,10 +97,268 @@ def test_connection_timeout(self, reqmock):
+            CloudFactoryDocumentRun.STATUS_TIMEOUT,
+            CloudFactoryDocumentRun.STATUS_ABORTED,
+        ]:
+            cf_run = CloudFactoryDocumentRun.objects.create(status=status, run_id="SOME_RUN")
+            reqmock.get(
+                settings.CLOUDFACTORY_API_URL + '/runs/' + cf_run.run_id,
+                status_code=200,
+                text='{"status": "Processed"}',
+            )
+            self.task(cf_run.id)
+            self.assertTrue(reqmock.called)
+            cf_run.refresh_from_db()
+            self.assertEqual(cf_run.status, CloudFactoryDocumentRun.STATUS_PROCESSED)
+            cf_run.delete()
+
+    def test_200_invalid(self, reqmock):
+        for status in [
+            CloudFactoryDocumentRun.STATUS_PROCESSING,
+            CloudFactoryDocumentRun.STATUS_TIMEOUT,
+            CloudFactoryDocumentRun.STATUS_ABORTED,
+        ]:
+            cf_run = CloudFactoryDocumentRun.objects.create(status=status, run_id="SOME_RUN")
+            reqmock.get(
+                settings.CLOUDFACTORY_API_URL + '/runs/' + cf_run.run_id,
+                status_code=200,
+                text='This is some invalid, non-json-parsable response content.',
+            )
+            self.assertRaises(ValueError, self.task, cf_run.id)
+            self.assertTrue(reqmock.called)
+            cf_run.refresh_from_db()
+            self.assertEqual(cf_run.status, CloudFactoryDocumentRun.STATUS_ERROR)
+            cf_run.delete()
+
+    def test_404_notfound(self, reqmock):
+        for status in [
+            CloudFactoryDocumentRun.STATUS_PROCESSING,
+            CloudFactoryDocumentRun.STATUS_TIMEOUT,
+            CloudFactoryDocumentRun.STATUS_ABORTED,
+        ]:
+            cf_run = CloudFactoryDocumentRun.objects.create(status=status, run_id="SOME_RUN")
+            reqmock.get(
+                settings.CLOUDFACTORY_API_URL + '/runs/' + cf_run.run_id, status_code=404, text='{}'
+            )
+            self.task(cf_run.id)
+            cf_run.refresh_from_db()
+            self.assertTrue(reqmock.called)
+            self.assertEqual(cf_run.status, CloudFactoryDocumentRun.STATUS_NOTFOUND)
+            cf_run.delete()
+
+    def test_other_response_status_code(self, reqmock):
+        initial_status = CloudFactoryDocumentRun.STATUS_PROCESSING
+        for status_code in [201, 202, 302, 400, 422, 500]:  # some common response statuses
+            cf_run = CloudFactoryDocumentRun.objects.create(
+                status=initial_status, run_id="SOME_RUN"
+            )
+            reqmock.get(
+                settings.CLOUDFACTORY_API_URL + '/runs/' + cf_run.run_id,
+                status_code=status_code,
+                text='{}',
+            )
+            with self.assertRaises(ValueError):
+                self.task(cf_run.id)
+            cf_run.refresh_from_db()
+            self.assertTrue(reqmock.called)
+            self.assertEqual(cf_run.status, initial_status)
+            cf_run.delete()
```

Comment:
don't need to `delete()` here as the TestCase will clean up created objects. (this applies to several other deletes in other test cases here as well.

Reply:
These runs are in a loop inside the test case method, so the TestCase won't clean them up for me, and I'll get errors for re-using the same run_id:
```
root@19d73ebf1f61:/hydroshare# python manage.py test myhpom.tests.test_cloudfactory_tasks
Creating test database for alias 'default'...
EEEEEE........EEEE
======================================================================
ERROR: test_202_invalid (myhpom.tests.test_cloudfactory_tasks.CloudFactoryAbortDocumentRunTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
...
DETAIL:  Key (run_id)=(SOME_RUN) already exists.
```

Reply:
It's needed since we're in a  loop.

Reply:
oh...b/c the run_id is the same. okay, disregard.